### ISSUE TITLE
[Question] Extra neural network defined inside actor_critic.py affect RL training reward?

### ISSUE THREAD

**Issue by nopaddleboat** [isaac-sim/IsaacLab#2798]
### Question

I am using isaac lab for humanoid locomotion RL training. I want to customize the rsl_rl, when I add an extra network in actor_critic.py as follows, the rewards are affected, even when the added netowork are never used.

 ```python
       mlp_input_dim_a = num_actor_obs
        mlp_input_dim_c = num_critic_obs
        # Policy
        actor_layers = []
        actor_layers.append(nn.Linear(mlp_input_dim_a, actor_hidden_dims[0]))
        actor_layers.append(activation)
        for layer_index in range(len(actor_hidden_dims)):
            if layer_index == len(actor_hidden_dims) - 1:
                actor_layers.append(nn.Linear(actor_hidden_dims[layer_index], num_actions))
            else:
                actor_layers.append(nn.Linear(actor_hidden_dims[layer_index], actor_hidden_dims[layer_index + 1]))
                actor_layers.append(activation)
        self.actor = nn.Sequential(*actor_layers)
        
        # Value function
        critic_layers = []
        critic_layers.append(nn.Linear(mlp_input_dim_c, critic_hidden_dims[0]))
        critic_layers.append(activation)
        for layer_index in range(len(critic_hidden_dims)):
            if layer_index == len(critic_hidden_dims) - 1:
                critic_layers.append(nn.Linear(critic_hidden_dims[layer_index], 1))
            else:
                critic_layers.append(nn.Linear(critic_hidden_dims[layer_index], critic_hidden_dims[layer_index + 1]))
                critic_layers.append(activation)
        self.critic = nn.Sequential(*critic_layers)

        print(f"Actor MLP: {self.actor}")
        print(f"Critic MLP: {self.critic}")

        # Added extra neural network, note that  self.est are never called
        est_layers = []
        est_layers.append(nn.Linear(mlp_input_dim_a, actor_hidden_dims[0]))
        est_layers.append(activation)
        for layer_index in range(len(actor_hidden_dims)):
            if layer_index == len(actor_hidden_dims) - 1:
                est_layers.append(nn.Linear(actor_hidden_dims[layer_index], num_actions))
            else:
                est_layers.append(nn.Linear(actor_hidden_dims[layer_index], actor_hidden_dims[layer_index + 1]))
                est_layers.append(activation)
        self.est = nn.Sequential(*est_layers)       
```
After the extra are added without other part changed, the reward is different. My understanding is that the self.est did not participate in the optimization if not called, why the rewards are affected? The following image shows the reward change, if I comment out the extra network, the reward curve with be the same every time I run the code. Am I missing something in this case, or is it a bug? 

![Image](https://github.com/user-attachments/assets/38fc83d8-879c-4466-ab7b-2ee0a9807fed)

**Comment by kellyguo11** [comment]
Hi there, we may be able to help better if you post this in the RSL RL github. do you see consistent behavior when running with different seeds without your changes? it could also be some stochasticity in the training setup.

**Comment by nopaddleboat** [comment]
Thanks for the reply. Yes, at first, I also doubt that it is  due to the stochasticity, but after the training steps go to 1000 or more, the difference is very significant as the following image shows. I will try to create a minimal working example. This question will be closed before the MWE is ready.

![Image](https://github.com/user-attachments/assets/7e31cca5-6700-499c-8505-d9e780a41ea4)
